Subject: [PATCH] build: replace deprecated Perl switch statements

Perl 5.38 deprecates given/when. Since the build scripts make all warnings
fatal, loading BuildCommon.pm fails during configure and gen-crypt-h fails
during compilation. Use equivalent if/elsif statements instead, preserving
first-match handling without depending on the experimental switch feature.

Upstream-Status: Pending

--- a/build-aux/scripts/BuildCommon.pm
+++ b/build-aux/scripts/BuildCommon.pm
@@ -519,19 +519,16 @@
     my $COMPAT_ABI;
     local $_;
     for (@args) {
-        when (/^SYMVER_MIN=(.+)$/) {
+        if (/^SYMVER_MIN=(.+)$/) {
             $usage_error->() if defined $SYMVER_MIN;
             $SYMVER_MIN = $1;
-        }
-        when (/^SYMVER_FLOOR=(.+)$/) {
+        } elsif (/^SYMVER_FLOOR=(.+)$/) {
             $usage_error->() if defined $SYMVER_FLOOR;
             $SYMVER_FLOOR = $1;
-        }
-        when (/^COMPAT_ABI=(.+)$/) {
+        } elsif (/^COMPAT_ABI=(.+)$/) {
             $usage_error->() if defined $COMPAT_ABI;
             $COMPAT_ABI = $1;
-        }
-        default {
+        } else {
             $usage_error->() if defined $map_in;
             $map_in = $_;
         }
--- a/build-aux/scripts/gen-crypt-h
+++ b/build-aux/scripts/gen-crypt-h
@@ -37,22 +37,16 @@
     local $_;
     while (<$fh>) {
         chomp;
-        # Yes, 'given $_' is really required here.
-        given ($_) {
-            when ('#define HAVE_SYS_CDEFS_H 1') {
-                $have_sys_cdefs_h = 1;
-            }
-            when ('#define HAVE_SYS_CDEFS_BEGIN_END_DECLS 1') {
-                $have_sys_cdefs_begin_end_decls = 1;
-            }
-            when ('#define HAVE_SYS_CDEFS_THROW 1') {
-                $have_sys_cdefs_throw = 1;
-            }
-            when (/^#define PACKAGE_VERSION "((\d+)\.(\d+)\.\d+)"$/) {
-                $substs{XCRYPT_VERSION_STR}   = $1;
-                $substs{XCRYPT_VERSION_MAJOR} = $2;
-                $substs{XCRYPT_VERSION_MINOR} = $3;
-            }
+        if ($_ eq '#define HAVE_SYS_CDEFS_H 1') {
+            $have_sys_cdefs_h = 1;
+        } elsif ($_ eq '#define HAVE_SYS_CDEFS_BEGIN_END_DECLS 1') {
+            $have_sys_cdefs_begin_end_decls = 1;
+        } elsif ($_ eq '#define HAVE_SYS_CDEFS_THROW 1') {
+            $have_sys_cdefs_throw = 1;
+        } elsif (/^#define PACKAGE_VERSION "((\d+)\.(\d+)\.\d+)"$/) {
+            $substs{XCRYPT_VERSION_STR}   = $1;
+            $substs{XCRYPT_VERSION_MAJOR} = $2;
+            $substs{XCRYPT_VERSION_MINOR} = $3;
         }
     }
 
